testapps/test_pillowtop/tests/test_case_search_pillow.py

```diff
@@ -108,7 +110,9 @@ def test_geopoint_property(self):
             'other_coords': '42 Wallaby Way',
             'not_coords': '-33.8561 151.2152 0 0',
         })
-        CaseSearchReindexerFactory(domain=self.domain).build().reindex()
+        with patch('corehq.pillows.case_search.domain_has_privilege') as mock_domain_has_privilege:
```

Comment:
I originally tried doing a simple `privilege_enabled()` decorator on this function, however I was unable to get that working correctly. Being unable to make the decorator work, I opted to simply patch out the `domain_has_privilege` function here.

Reply:
Ya patching is tricky at times.
I would be game to debug this so if you can let me know to replicate, I can try to see how to use `privilege_enabled`

Reply:
Sure! To replicate this you can simply remove the code from the  [272bcce](https://github.com/dimagi/commcare-hq/pull/35001/commits/272bcce54f2d20ea06615614c9817df0d0be941d) commit and instead add a `privilege_enabled()` decorator for the test function. When testing this it seemed the issue was that even when the `privilege_enabled()` decorator was added, the `domain_has_privilege` check which was added in the [128c8e7](https://github.com/dimagi/commcare-hq/pull/35001/commits/128c8e7c3c5bcd087dfd2cd8ca61dba5cb7ee513) commit would still fail.

Reply:
Hey @mkangia, Ajeet has pointed out a solution for resolving the above issue. See [this](https://github.com/dimagi/commcare-hq/pull/35001#discussion_r1731122205) comment for more context.